### PR TITLE
Move from Lightsail deployment to EC2, automatic CloudFormation setup (w/beta env)

### PR DESCRIPTION
**Pull request by mathcolo**
This is nearly an exact copy of the RRE architecture I put together over in https://github.com/transitmatters/regional-rail-explorer/pull/50

Please consult the documentation at https://docs.google.com/document/d/1WjCsdzGlIQjt26CYvinbA2-yIXOAuxmTz8GYgOYS3MI/edit if you have any questions about it works.

### CODE REVIEW

File: devops/ansible.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+- name: deploy latest new train tracker on EC2
+  hosts: all
+  remote_user: ubuntu
+  become_user: ubuntu
+  tasks:
+    - name: "Add nodejs apt key"
```
Comment (nathan-weinberg) [inline devops/ansible.yml:7]:
```suggestion
    - name: add nodejs apt key
```

File: devops/ansible.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      become: yes
+      become_user: root
+      apt_key:
+        url: https://deb.nodesource.com/gpgkey/nodesource.gpg.key
+        state: present
+
+    - name: "Add nodejs 16.x ppa for apt repo"
```
Comment (nathan-weinberg) [inline devops/ansible.yml:14]:
```suggestion
    - name: add nodejs 16.x ppa for apt repo
```

File: devops/ansible.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
+---
```
Comment (nathan-weinberg) [inline devops/ansible.yml:1]: I don't love the name of this - could we do something like `deploy-playbook.yml` instead? I missed this on RRE so maybe we could update it there too?

File: devops/ansible.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        state: present
+
+    - name: "Add nodejs 16.x ppa for apt repo"
+      become: yes
+      become_user: root
+      apt_repository:
+        repo: deb https://deb.nodesource.com/node_16.x focal main
+        update_cache: yes
+
+    - name: Install nodejs
```
Comment (nathan-weinberg) [inline devops/ansible.yml:21]:
```suggestion
    - name: install nodejs
```

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -8,17 +8,31 @@ on:
...
+          [transitmatters]
+          aws_access_key_id = ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws_secret_access_key = ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          EOF
+
+          mkdir ~/.ssh
+          cat >> ~/.ssh/transitmatters-ntt.pem << EOF
+          ${{ secrets.SSH_PRIVATE_KEY }}
+          EOF
+          chmod 0600 ~/.ssh/transitmatters-ntt.pem
```
Comment (nathan-weinberg) [inline .github/workflows/deploy.yml:34]: I feel like maybe this should be a separate task?

File: devops/ansible.yml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      become: yes
+      become_user: root
+      ansible.builtin.apt:
```
Comment (nathan-weinberg) [inline devops/ansible.yml:24]: This should work with just `apt` AFAIK

File: devops/ansible.yml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      become: yes
+      become_user: root
+      copy:
+        src: systemd.conf
+        dest: /etc/systemd/system/new-train-tracker.service
+
+    - name: Install nginx
+      become: yes
+      become_user: root
+      ansible.builtin.apt:
```
Comment (nathan-weinberg) [inline devops/ansible.yml:47]: Same comment as above

File: devops/deploy.sh [hunk 8]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+set -e
```
Comment (nathan-weinberg) [inline devops/deploy.sh:2]:
```suggestion
#!/bin/bash

set -e
```

File: docs/README.md [hunk 9]
```diff
@@ -11,20 +11,14 @@ Dependencies:
 - `node` `16.x`
 - `python` `3.8`
 - [`poetry`](https://python-poetry.org/)
-- [`postgresql`](https://www.postgresql.org/)
 
 Run:
 - `npm install`
-- `npm run create-history-db` if it is your first time running the application, `npm run update-history-db` otherwise
 - `npm start`
 - visit [http://localhost:5000/](http://localhost:5000/)
```
Comment (nathan-weinberg) [inline docs/README.md:18]: Deployment steps lower in the README need to be updated as well

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]

File: devops/cloudformation.json [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+    "NTTCertArn": {
+      "Type": "String",
+      "Description": "The ACM ARN of the SSL certificate."
+    }
+  },
+  "Resources": {
+    "NTTInstance": {
+      "Type": "AWS::EC2::Instance",
+      "Properties": {
+        "ImageId": "ami-070650c005cce4203",
```
Comment (thiteixeira) [inline devops/cloudformation.json:24]: Does this image change often? If so, do we have a workflow to automatically update and upload them?

File: poetry.lock [hunk 11]
```diff
@@ -320,11 +320,11 @@ python-versions = "*"
 
 [[package]]
 name = "pathspec"
-version = "0.9.0"
+version = "0.10.1"
 description = "Utility library for gitignore style pattern matching of file paths."
 category = "dev"
 optional = false
-python-versions = "!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,>=2.7"
+python-versions = ">=3.7"
```
Comment (thiteixeira) [inline poetry.lock:327]: Nit: The [lint workflow](https://github.com/transitmatters/new-train-tracker/blob/master/.github/workflows/lint.yml#L14) is using Python 3.8. Should these two match?

**Review by thiteixeira — COMMENTED** [review]
Good stuff. Added a couple comments

File: poetry.lock (hunk 11 above)
Comment (devinmatte) [reply poetry.lock:327]: I would agree that we should pin to at minimum 3.8, no reason to run on anything lower

File: devops/cloudformation.json [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+      "Type": "String",
+      "Default": "ntt-beta.labs.transitmatters.org",
+      "AllowedPattern": "^traintracker\\.transitmatters\\.org$|^.*\\.labs\\.transitmatters\\.org$",
+      "Description": "The hostname for the site; for example: traintracker.transitmatters.org."
+    },
+    "NTTDomain": {
+      "Type": "String",
+      "Default": "labs.transitmatters.org",
+      "AllowedPattern": "^traintracker\\.transitmatters\\.org$|^labs\\.transitmatters\\.org$",
+      "Description": "The parent domain of the site. Most likely traintracker.transitmatters.org or labs.transitmatters.org."
```
Comment (Joshhw) [inline devops/cloudformation.json:13]: what's the need for both of these? they seem to be the same.

**Review by Joshhw — APPROVED** [review]
This all seems fine to me. Im not used to seeing deployments on actual servers rather than docker/k8s.

**Comment by nathan-weinberg** [comment]
I think we can close https://github.com/transitmatters/new-train-tracker/issues/41 when this merges

**Pull request merged**